power-sequencer/mihawk-cpld.cpp

```diff
@@ -596,5 +682,65 @@ void MIHAWKCPLD::clearCPLDregister()
     close(fd);
 }
 
+// Check for Mowgli's HDDFault
+int MIHAWKCPLD::checkFault(int reg)
+{
+    int result = 0;
+    std::string i2cBus = "/dev/i2c-" + std::to_string(busId);
+
+    // open i2c device(CPLD-PSU-register table)
+    int fd = open(i2cBus.c_str(), O_RDWR | O_CLOEXEC);
+    if (fd < 0)
+    {
+        std::cerr << "Unable to open i2c device \n";
+    }
+
+    // set i2c slave address
+    if (ioctl(fd, I2C_SLAVE_FORCE, slaveAddr) < 0)
+    {
+        std::cerr << "Unable to set device address \n";
+        close(fd);
```

Comment:
Is there any point in continuing here since you're closing the FD anyway?  Could just return 0; after all of these failures?

Reply:
Done.